cli_tool/main.cpp

```diff
@@ -0,0 +1,68 @@
+#include <iostream>
+#include <gflags/gflags.h>
+#include <string>
+
+// Defining all Command Line flags here 
+DEFINE_string(registeruser, "", "Registers the given username");
+DEFINE_string(user, "", "Logs in as the given username");
+DEFINE_string(caw, "", "Creates a new Caw with the given text");
+DEFINE_int64(reply, -1, "Indicates that the new caw is a reply to the given id");
+DEFINE_string(follow, "", "Starts following the given username");
+DEFINE_int64(read, -1, "Reads the caw thread starting at the given id");
+DEFINE_bool(profile, false, "Gets the user’s profile of following and followers");
+
+int main(int argc, char* argv[]){
+
+	// Checking and handeling arguments 
+	if(argc < 2){
+		std::cout << "You did not specify any program functions." << std::endl;
```

Comment:
This kind of help text can be auto-generated by gflags so it's not necessary.

Reply:
I wrote it here because gflags included a lot of irrelevant data in a hard to read format (help flags etc.) Do you think I should display all the information instead?